nsiqcppstyle_rulemanager.py

```diff
@@ -101,6 +104,13 @@ def RunPreprocessRule(self, lexer, contextStack):
             preprocessRule(lexer, contextStack)
             lexer.Restore(data)
 
+    def RunCommentRule(self, lexer, token):

```

Comment:
In a future PR, let's see if we can reduce the duplication by:
```py
    def RunPreprocessRule(self, lexer: Lexer, contextStack: ??) -> None:
        """ Run rules which runs in the preprecessor blocks """
        _RunLexerRules(self.preprocessRules, lexer, contextStack)

    def _RunLexerRules(self, rules: List[Callable], lexer,  *args, **kwargs) -> None
        for rule in rules:
            data = lexer.Backup()
            rule(lexer, *args, **kwargs)
            lexer.Restore(data)
```

Reply:
Definitely a concise form of invoking all lexer-related rules.  Note that:

- def RunSessionEndRules(self):
- def RunSessionStartRules(self):
- def RunProjectRules(self, targetName):

will have alternative code for invoking them.  By all means, open an issue to track this.